modules/build-tarball.nix

```
@@ -49,14 +49,17 @@ let
     # Write wsl.conf so that it is present when NixOS is started for the first time
     cp ${config.environment.etc."wsl.conf".source} ./etc/wsl.conf
 
-    ${lib.optionalString config.wsl.tarball.includeConfig ''
+    ${lib.optionalString config.wsl.tarball.includeConfig (if config.wsl.tarball.configPath == null then ''
       # Copy the system configuration
       mkdir -p ./etc/nixos/nixos-wsl
       cp -R ${lib.cleanSource ../.}/. ./etc/nixos/nixos-wsl
       mv ./etc/nixos/nixos-wsl/configuration.nix ./etc/nixos/configuration.nix
       # Patch the import path to avoid having a flake.nix in /etc/nixos
       sed -i 's|import \./default\.nix|import \./nixos-wsl|' ./etc/nixos/configuration.nix
-    ''}
+    '' else ''
+      mkdir -p ./etc/nixos
+      cp -R ${lib.cleanSource config.wsl.tarball.configPath}/. ./etc/nixos
+    '')}
   '';
```

```suggestion
    '');
```
didn't test but instead of inlining the string my idea is to append it to reduce the nested things here